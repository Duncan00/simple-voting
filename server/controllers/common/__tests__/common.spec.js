'use strict';

const initServer = () => require('../../../initServer')();
const supertest = require('supertest');

describe('Common controller', () => {
	let server;
	beforeEach(async () => {
		jest.resetModules();
		server = await initServer();
	});
	afterEach(() => {
		server.close();
	});

	describe('whoami', () => {
		test('successfully call GET', async () => {
			const response = await supertest(server).get('/v1/whoami');
			expect(response.status).toEqual(200);
			expect(response.body).toMatchSnapshot('GET whoami success');
		});
	});
});
