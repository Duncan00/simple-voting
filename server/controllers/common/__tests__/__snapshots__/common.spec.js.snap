// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common controller whoami successfully call GET: GET whoami success 1`] = `
Object {
  "data": Object {
    "build": "Unknown",
    "service": "simple-voting",
    "version": "v1",
  },
  "meta": Object {
    "code": 20000,
    "message": "OK",
  },
}
`;
